persimmon/website/views.py

```diff
@@ -133,3 +144,49 @@ def get_my_accounts(request):
         'balance': account.balance,
         'approval_status': account.approval_status,
     } for account in accounts]
+
+
+@api_function
+def approve_credit_debit_funds(request, account_number: int, approved: bool, credittype0debittype1: int):
+    current_user(request, required_auth=EmployeeLevel.TELLER)
+    try:
+        pendingtransactions = BankStatements \
+            .objects.filter(bankAccountId=account_number, approval_status=ApprovalStatus.PENDING)
+    except BankStatements.DoesNotExist as exc:
+        raise Http404("No such transaction pending approval") from exc
+    for creditdebit in pendingtransactions:
+        if approved:
+            creditdebit.approve_status = ApprovalStatus.APPROVED
+            account = BankAccount.objects.filter(id=account_number)
+            if credittype0debittype1 == 0:
+                account.balance -= int(creditdebit.transaction[1])
+            else:
+                account.balance += int(creditdebit.transaction[1])
+        else:
+            creditdebit.approve_status = ApprovalStatus.DECLINED
+        now = datetime.now()
+        creditdebit.date = now.strftime("%d %b")
+        creditdebit.save()
+
+
+@api_function
+def credit_debit_funds(request, account_number: int, balance: Decimal, credittype0debittype1: int):
+    current_user(request)
+    if credittype0debittype1 == 0:
+        transactionbalance = "-" + balance
+    else:
+        transactionbalance = "+" + balance
```

Comment:
Does it not work to just pass a positive or negative value for balance instead of specifying a separate field for positive or negative?

Reply:
oh yeah, that would work , what I thought before was to change it to a string.